src/main/frontend/components/query_table.cljs

```diff
@@ -77,19 +49,73 @@
     :else
     true))
 
+(defn- get-sort-state
+  "Return current sort direction and column (item) being sorted, respectively
+  :sort-desc? and :sort-by-item. :sort-by-item is nil if no sorting is to be
+  done"
+  [state current-block]
+  (let [*sort-by-item (get state ::sort-by-item)
+        *desc? (get state ::desc?)
+        p-desc? (get-in current-block [:block/properties :query-sort-desc])
+        desc? (desc? *desc? p-desc?)
+        p-sort-by (keyword (get-in current-block [:block/properties :query-sort-by]))
+        sort-by-item (or @*sort-by-item
+                         (some-> p-sort-by keyword)
+                         (if (query-dsl/query-contains-filter? (:block/content current-block) "sort-by")
+                           nil
+                           :updated-at))]
+    {:sort-desc? desc?
+     :sort-by-item sort-by-item}))
+
+(defn- sort-result [result {:keys [sort-by-item sort-desc?]}]
+  (if (some? sort-by-item)
+    (let [comp (if sort-desc? > <)]
+      (sort-by (fn [item]
+                 (block/normalize-block (sort-by-fn sort-by-item item)))
+               comp
+               result))
+    result))
+
+;; Components and public fns
+;; =========================
+(rum/defc sortable-title
+  [title key state {:keys [sort-by-item sort-desc?]} block-id]
+  (let [*sort-by-item (get state ::sort-by-item)
+        *desc? (get state ::desc?)]
+    [:th.whitespace-nowrap
+     [:a {:on-click (fn []
+                      ;; The two local state atom changes have no effect on
```

Comment:
@tiensonqin When I disable the two local state atoms for simple queries, it has no effect on preserving sorting state as that is handled by the block property. Do we know what local state is needed for here? The `block/*custom-query` component is used in so many contexts that I left the local atoms in order to not accidentally break something

Reply:
I think they're not needed anymore because updating either :query-sort-by or :query-sort-desc will trigger react/refresh! which will re-run the query and rerender the component.

Reply:
Thanks. Makes sense. Will remove